src/shared/csv.ts

```diff
@@ -1,13 +1,22 @@
+const CSV_REGEX_CHARS = {
+  comma: ',',
+  tab: '\\t',
+} as const satisfies Record<string, string>;
+
+export type CSVSeparator = keyof typeof CSV_REGEX_CHARS;
```

Comment:
This feels a bit elaborate to me. Any reason not to use just `type CSVSeparator = ',' | '\\t'`?

Reply:
I thought passing `\t` might feel a bit arbitrary, and using character names would be more clear. Also, it needs to be escaped for the regexp, but not when used as joining character.

That said, I agree it is a bit too elaborate and I tend to agree we should be more pragmatic here, so let me simplify it.

Reply:
> Also, it needs to be escaped for the regexp, but not when used as joining character.

Ok, this is actually not correct. I was doing one extra escaping which is not needed, so one more reason in favor of simplifying this.